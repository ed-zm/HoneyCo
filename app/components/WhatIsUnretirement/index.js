import React from 'react'
import styled from 'styled-components'
import { Container } from 'reactstrap'
import { HoneyCoBlueRGBA, HoneyCoRed, HoneyCoDarkGray, HoneyCoYellow } from '../../constants'

const StyledContainer = styled.div`
  height: 768px;
  width: 100%;
  background: url(${(props) => props.background});
  background-position: center;
  background-size: cover;
`

const StyledContainerTransparency = styled.div`
  height: 768px;
  width: 100%;
  background-color: ${HoneyCoBlueRGBA(0.6)};
  display: flex;
  align-items: flex-start;
  justify-content: flex-start;
  padding-top: 3em;
`

const StyledTitle = styled.span`
  display: block;
  font-size: 50px;
  color: ${HoneyCoRed};
  margin-bottom: 0.5em;
`

const StyledParagraph = styled.p`
  width: 60%;
  font-size: 30px;
  color: white;
  margin-bottom: 1em;
`

const StyledButton = styled.button`
  font-family: Oswald;
  font-size: 26px;
  background-color: ${HoneyCoYellow};
  color: ${HoneyCoDarkGray};
  padding: .2em;
  border-radius: 5px;
  border: none;
  padding: .3em 1.5em;
  text-align: center;
  margin-top: 10px;
`

const WhatIsUnretirement = (props) =>
  <StyledContainer background = {props.background}>
    <StyledContainerTransparency>
      <Container>
        <StyledTitle>{props.text.title}</StyledTitle>
        <StyledParagraph>{props.text.firstParagraph}</StyledParagraph>
        <StyledParagraph>{props.text.secondParagraph}</StyledParagraph>
        <StyledButton>{props.text.button}</StyledButton>
      </Container>
    </StyledContainerTransparency>
  </StyledContainer>

export default WhatIsUnretirement
