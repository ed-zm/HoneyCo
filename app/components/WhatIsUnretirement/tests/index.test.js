import React from 'react'
import { shallow } from 'enzyme'
import WhatIsUnretirement from '../index'

const text = {
  title: 'title',
  firstParagraph: 'first paragraph',
  secondParagraph: 'second paragraph',
}
describe('<WhatIsUnretirement />', () => {
  it('It should render the WhatIsUnretirement Component', () => {
    const wrapper = shallow(<WhatIsUnretirement text = {text}/>)
    expect(wrapper).toMatchSnapshot()
  })
})
