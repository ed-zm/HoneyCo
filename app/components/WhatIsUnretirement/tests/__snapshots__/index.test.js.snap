// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WhatIsUnretirement /> It should render the WhatIsUnretirement Component 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <styled.div
    background={undefined}
>
    <styled.div>
        <gt
            tag="div"
        >
            <styled.span>
                title
            </styled.span>
            <styled.p>
                first paragraph
            </styled.p>
            <styled.p>
                second paragraph
            </styled.p>
            <styled.button />
        </gt>
    </styled.div>
</styled.div>,
  "nodes": Array [
    <styled.div
      background={undefined}
>
      <styled.div>
            <gt
                  tag="div"
            >
                  <styled.span>
                        title
                  </styled.span>
                  <styled.p>
                        first paragraph
                  </styled.p>
                  <styled.p>
                        second paragraph
                  </styled.p>
                  <styled.button />
            </gt>
      </styled.div>
</styled.div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <WhatIsUnretirement
        text={
                Object {
                        "firstParagraph": "first paragraph",
                        "secondParagraph": "second paragraph",
                        "title": "title",
                      }
        }
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "text": Object {
            "firstParagraph": "first paragraph",
            "secondParagraph": "second paragraph",
            "title": "title",
          },
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <styled.div
          background={undefined}
>
          <styled.div>
                    <gt
                              tag="div"
                    >
                              <styled.span>
                                        title
                              </styled.span>
                              <styled.p>
                                        first paragraph
                              </styled.p>
                              <styled.p>
                                        second paragraph
                              </styled.p>
                              <styled.button />
                    </gt>
          </styled.div>
</styled.div>,
        "_debugID": 2,
        "_renderedOutput": <styled.div
          background={undefined}
>
          <styled.div>
                    <gt
                              tag="div"
                    >
                              <styled.span>
                                        title
                              </styled.span>
                              <styled.p>
                                        first paragraph
                              </styled.p>
                              <styled.p>
                                        second paragraph
                              </styled.p>
                              <styled.button />
                    </gt>
          </styled.div>
</styled.div>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <WhatIsUnretirement
    text={
        Object {
            "firstParagraph": "first paragraph",
            "secondParagraph": "second paragraph",
            "title": "title",
          }
    }
/>,
}
`;
